lightning/src/routing/router.rs

```diff
@@ -912,14 +916,20 @@ where L::Target: Logger {
 					let over_path_minimum_msat = amount_to_transfer_over_msat >= $candidate.htlc_minimum_msat() &&
 						amount_to_transfer_over_msat >= $next_hops_path_htlc_minimum_msat;
 
+					#[allow(unused_comparisons)] // $next_hops_path_htlc_minimum_msat is 0 in some calls so rustc complains
+					let may_overpay_to_meet_path_minimum_msat =
+						((amount_to_transfer_over_msat < $candidate.htlc_minimum_msat() &&
```

Comment:
this is a rather convoluted comparison. Do you think something like what you do in line 930, having each boolean value be assigned to a variable (maybe with some comments for each one) would improve legibility?

Reply:
I imagine it can be broken up into the two variables where one is `$candidate.htlc_minimum_msat()` falls between `amount_to_transfer_over_msat ` and `recommended_value_msat` and the other is the same but for `$next_hops_path_htlc_minimum_msat`. Do you have any recommendations on naming either?

Reply:
is_htlc_minimum_msat_within_tolerance_range perhaps?

Reply:
Hmm, that doesn't seem any more clear to me - its not a "tolerance" range more of a "well, its within the range where we should try to meet it instead of ignoring the channel" range.

Reply:
well, nomenclature aside, the logic LGTM, so I'm gonna approve and then re-ack if we wanna change this part

Reply:
Lets land it - its blocking release and its probably clearer than it was given there's now some comment verbiage lol.